inc/auth.php

```diff
@@ -468,7 +468,13 @@ function auth_ismanager($user = null, $groups = null, $adminonly = false, $recac
         }
     }
     if(is_null($groups)) {
-        $groups = $USERINFO ? (array) $USERINFO['grps'] : array();
+        if ($user === $INPUT->server->str('REMOTE_USER')) {
+            $groups = $USERINFO ? (array) $USERINFO['grps'] : array();
```

Comment:
This still does not seem totally right. When there is not `$USERINFO` you will return an empty array instead of fetching the groups via `getUserData`. Admittedly this will probably never happen, but if you check for `$USERINFO` then the fallback should be correct.

Overall I'm not too happy about all these ifs and ternary operators. Is there a way to reformat this code in a way that has less nested conditionals?

Reply:
This is the most straightforward condition I can come up with:

```php
    if(is_null($groups)) {
        // checking the logged in user, or another one?
        if ($USERINFO && $user === $INPUT->server->str('REMOTE_USER')) {
            $groups =  (array) $USERINFO['grps'];
        } else {
            $groups = (array) $auth->getUserData($user)['grps'];
        }
    }
```

Now I have to figure out what is the best way to update the old tests (`AuthPlugin` does not have a real implementation of `getUserData()` so some tests will fail).

Reply:
```$auth->getUserData($user)['grps'];``` will fail when getUserData can't find the user and returns an empty array.

Which tests are failing? Can they be rewritten to test against auth_plain instead of AuthPlugin?

Reply:
`auth->getUserData($user)` will return `false`, not an empty array, when the method is not implemented.

This results in multiple tests failing with `Argument 3 passed to auth_isMember() must be of the type array, null given`

When I cast the return value into an (empty) array, as in the latest commit https://github.com/splitbrain/dokuwiki/pull/3318/commits/1525c2281e6bc28f12ce8a59976e68e5a0e788fa , the tests pass and my PR should be fit for merging.